src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -1432,11 +1502,10 @@ public int doFinal(ByteBuffer src, ByteBuffer dst)
 
             checkDataLength(len);
 
-            // Save GHASH context to allow the tag to be checked even though
-            // the dst buffer is too short.  Context will be restored so the
-            // method can be called again with the proper sized dst buffer.
+            // Verify dst is large enough
             if (len > dst.remaining()) {
-                save = ghashAllToS.clone();
+                throw new ShortBufferException("Output buffer too small, must" +
+                    "be at least " + (len - tagLenBytes) + " bytes long");
```

Comment:
nit: add a space between "must" and "be".
Instead of '(len -tagLenBytes)' should just be 'len' as its initial value already deducts tagLenBytes (see line 1495).